rtpreceiver.go

```diff
@@ -411,13 +418,42 @@ func (r *RTPReceiver) receiveForRtx(ssrc SSRC, rsid string, streamInfo *intercep
 	track.repairInterceptor = rtpInterceptor
 	track.repairRtcpReadStream = rtcpReadStream
 	track.repairRtcpInterceptor = rtcpInterceptor
+	track.repairStreamChannel = make(chan rtxPacketWithAttributes)
 
 	go func() {
 		b := make([]byte, r.api.settingEngine.getReceiveMTU())
 		for {
-			if _, _, readErr := track.repairInterceptor.Read(b, nil); readErr != nil {
+			i, attributes, err := track.repairInterceptor.Read(b, nil)
+			if err != nil {
+				return
+			}
+
+			pkt := &rtp.Packet{}
+			if err := pkt.Unmarshal(b[:i]); err != nil {
 				return
 			}
+
+			if len(pkt.Payload) < 2 {
+				// BWE probe packet, ignore
+				continue
+			}
+
+			// RTX packets have a different payload format. Move the OSN in the payload to the RTP header and rewrite the
+			// payload type and SSRC, so that we can return RTX packets to the caller 'transparently' i.e. in the same format
+			// as non-RTX RTP packets
+			attributes.Set("rtx_payload_type", pkt.Header.PayloadType)
+			attributes.Set("rtx_ssrc", pkt.Header.SSRC)
+			attributes.Set("rtx_sequence_number", pkt.Header.SequenceNumber)
```

Comment:
Perhaps define some exported constants for the fields in the attributes? They are effectively part of the API, and that would be more clear there were exports.

Reply:
Good idea, I'll make the change.